app/models/lesson.rb

```diff
@@ -0,0 +1,5 @@
+class Lesson < ApplicationRecord
+  belongs_to :topic
+  has_many :user_lessons
```

Comment:
I think 

`has_many :users`

or 

`has_and_belongs_to_many :users` 

[may be even better ](https://www.sitepoint.com/master-many-to-many-associations-with-activerecord/)

> ex. usage: user.lessons.where(completed: true) 
to grab all lessons a user has completed 
> ex. usage: lesson.users.thumbnail 
to grab thumbnails of all users who have started a lesson, so that a lesson.user can see who's working on the same thing in the lesson's show view